lib/xandra/connection/utils.ex

```diff
@@ -48,12 +48,30 @@ defmodule Xandra.Connection.Utils do
     # However, we need to pass the compressor module around when we
     # receive the response to this frame because if we said we want to use
     # compression, this response is already compressed.
-    with :ok <- :gen_tcp.send(socket, payload),
-         {:ok, %Frame{body: <<>>}} <- recv_frame(socket, compressor) do
-       :ok
+    with :ok <- :gen_tcp.send(socket, payload) do
+      case recv_frame(socket, compressor) do
+        {:ok, %Frame{body: <<>>}} -> :ok
+        {:ok, %Frame{kind: :authenticate}} ->
+          authenticate_connection(socket, requested_options, compressor, options)
+      end
```

Comment:
Using `case` here changes the semantics: before, we handled the non-`{:ok, _}` cases by using the `with` (which would break the chain and call the `else` part of the `with`). This way, if `recv_frame/2` fails (and it will fail because it's a network operation) we get a `CaseClauseError`. Please use something like

```elixir
with :ok <- :gen_tcp.send(...),
     {:ok, frame} <- recv_frame(socket, compressor) do
  case frame do
    %Frame{body: <<>>} -> ...
    %Frame{kind: :authenticate} -> ...
  end
# same as before
...
end
```